doc/cabal-package-description-file.rst

```diff
@@ -2693,46 +2693,68 @@ Starting with Cabal-2.2 it's possible to use common build info stanzas.
     :since: 1.6
 
-A marker that points to the *source code* for this package within a
-**source code repository**.
+A metadata marker that helps users of your package find its source code.
+This marker points to a *source code* snapshot within a
+version control systems (VCS) **repository** like Git.
 
-There are two kinds. You can specify one or the other or both at once:
+Here are some examples that point to snapshots of the Cabal library in
+its Git repository:
 
--  The ``head`` kind refers to the latest development branch of the
-   package. This may be used for example to track activity of a project
-   or as an indication to outside developers what sources to get for
-   making new contributions.
-
--  The ``this`` kind refers to the branch and tag of a repository that
-   contains the sources for this version or release of a package.  For most
-   source control systems this involves specifying a tag, id or hash of some
-   form and perhaps a branch.
+::
 
-As an example, here are the repositories for the Cabal library. Note that the
-``this`` kind of repository specifies a tag.
+    -- Snapshot via Git commit hash
+    source-repository this
+      type:     git
+      location: https://github.com/haskell/cabal
+      tag: 260ecdc3d848782d4df49e629cb0a5dc9e96ca9e
 
-::
+    -- Snapshot via Git tag
+    source-repository this
+      type:     git
+      location: https://github.com/haskell/cabal
+      tag:      cabal-install-v3.12.1.0
 
+    -- Snapshot via latest commit (head) of the default (master) branch
     source-repository head
       type:     git
       location: https://github.com/haskell/cabal
 
-    source-repository this
+    -- Snapshot via latest commit (head) of the specified branch
+    source-repository head
       type:     git
       location: https://github.com/haskell/cabal
-      tag:      1.6.1
+      branch:   3.12
+
+    -- Snapshot via latest commit (head) in subdirectory of a monorepo
+    source-repository head
+      type:     git
+      location: https://github.com/haskell/vector.git
+      subdir:   vector
+
+Note the two notation variants ``this`` and ``head``:
+
+-  The ``this`` variants refers to a specific snapshot of a repository
+   that contains the sources for this version or release of your package.
+   For most source control systems this involves specifying a tag, id or
+   hash of some commit and perhaps a branch.
+
+-  The ``head`` variant refers to the latest source code snapshot
+   in a repository branch of your package. This may be used for example
+   to track activity of a project or as an indication to outside
+   developers what sources to get for making new contributions.
 
 The :ref:`cabal get<cabal-get>` command uses the kind of repository with
 its ``--source-repository`` option, if provided.
 
 .. _source-repository-fields:
 
-The :ref:`VCS fields<vcs-fields>` of ``source-repository`` are:
+The fields of ``source-repository`` are:
 
 ..
+  -- Cabal internal datatype
```

Comment:
TBH I'm not sure we should be throwing Haskell data types at users like this.

Reply:
Removed